src/assets/TokenRatesController.ts

```diff
@@ -170,21 +214,43 @@ export class TokenRatesController extends BaseController<
    * @returns Promise resolving when this operation completes
    */
   async updateExchangeRates() {
-    if (this.tokenList.length === 0) {
+    if (this.tokenList.length === 0 || this.disabled) {
```

Comment:
If this was called directly, nothing was stopping it from being run. What is the intended behavior originally: should disable just disable tokenList/chainId changes?

Reply:
I don't think the `disabled` property was used in this controller :thinking:

Like many aspects of the BaseController, some controllers are written to use it and some ignore it, or break when you try to use it. The config is similar - it's built to be changeable at runtime, but many controllers are written assuming that never happens, and they break if you try it.

It doesn't hurt to add a check for `disabled` here but I'm not sure it helps either, unless I'm missing something. This awkwardness with the API is one of the things BaseControllerV2 is meant to address.